lib/utm_cookie.coffee

```diff
@@ -70,9 +74,10 @@ class UtmCookie
     regex = new RegExp(regexS)
     results = regex.exec(window.location.search)
     if results
-      decodeURIComponent results[1].replace(/\+/g, ' ')
+      value = decodeURIComponent results[1].replace(/\+/g, ' ')
+      if value.length > 0 then value else null
     else
-      ''
+      null
```

Comment:
Can you please elaborate the advantage of `null` over an empty string?

Reply:
It's required for initial parameters (otherwise they will be overwritten) and also for the "last-even-null" logic.

Reply:
To be honest, I'm still not sure about this. Since this method returns value to be written into the cookies, they'll be string with value "null" which seems incorrect. If there's no value, the cookie value should be an empty string.

Are you seeing something different for this case?

Reply:
It is known that empty cookie values may not behave the same way on all browsers.

In our specific case an empty value will also fail the existingValue check inside writeCookieOnce method.

So what would you suggest?

Reply:
That's a good point. In that case `writeCookie` needs to be updated to write only when `value` is a non-empty string. Similarly `readCookie` should return an empty string if no cookie is found. 

This might be a broader change that might be out of scope of your current goal. I can make these changes when I get the opportunity and also incorporate the changes proposed in this PR.

Reply:
On second thought, `readCookie` should return `null`, but when using the value to fill the fields, it could be an empty string when the value is `null`. This way the role of these methods is clearer.

What do you think?

Reply:
No problem, i can make the changes. But what about my first sentence? Do you still prefer to keep going with an empty cookie value?

Reply:
No, we should change that. You are right that the cookie value shouldn't be empty. But returning `null` here might not work since `writeUtmCookieFromParams` and other such methods write the cookie without checking if the value is null or not. So `null` value will be written as "null" string in the cookie (at least that's what I see in my basic testing).

Reply:
Cookie values are actually just strings. So a "real" null value, int, etc. cannot be stored inside cookies. So i think we should only choose a conventional value to identify a cookie without arbitrary data. Initially I thought about `null` or `0`, and proceeded with null.

Also cookie data should be encoded (like Base64) as for [RFC 6265](https://tools.ietf.org/html/rfc6265)

Reply:
Yes, I do realize that a real null value cannot be stored in cookies. My main concern with the changes is that if say `utm_campaign` is not present in the URL, then the form will be populated with UTM campaign as "null". It should have it as '' (empty string).

Do you agree that with these changes, "null" string will be written in cookies and the form fields will have "null" for any UTM value that is missing in the URL? If not, then maybe I have misunderstood something.

Reply:
I absolutely agree with you. The form fields are actually populated with `"null"` strings. They should be filled with `""` strings instead.

This PR is not ready to be merged. I'm just waiting your confirmations about the changes to do. As i said, i can make all the changes. So, let's choose our "conventions" and desired state and i will then start working on it.

Reply:
Yes, that works for me. This is the only concern I had regarding these changes.